std-bits/table/src/main/java/org/enso/table/data/column/operation/map/GenericBinaryObjectMapOperation.java

```diff
@@ -2,30 +2,37 @@
 
 import org.enso.base.polyglot.Polyglot_Utils;
 import org.enso.table.data.column.builder.Builder;
+import org.enso.table.data.column.builder.ObjectBuilder;
 import org.enso.table.data.column.storage.Storage;
 import org.enso.table.data.column.storage.type.AnyObjectType;
 import org.enso.table.error.UnexpectedTypeException;
 import org.graalvm.polyglot.Context;
 
 public abstract class GenericBinaryObjectMapOperation<
-        InputType, InputStorageType extends Storage<InputType>, OutputType>
+        InputType,
+        InputStorageType extends Storage<InputType>,
+        ArgType,
+        ArgStorageType extends Storage<ArgType>,
+        OutputType>
     extends BinaryMapOperation<InputType, InputStorageType> {
 
   protected GenericBinaryObjectMapOperation(
       String name,
-      Class<InputType> inputTypeClass,
-      Class<? extends InputStorageType> inputStorageTypeClass) {
+      Class<ArgType> argTypeClass,
+      Class<? extends ArgStorageType> argStorageTypeClass) {
     super(name);
-    this.inputTypeClass = inputTypeClass;
-    this.inputStorageTypeClass = inputStorageTypeClass;
+    this.argTypeClass = argTypeClass;
+    this.argStorageTypeClass = argStorageTypeClass;
   }
 
-  private final Class<InputType> inputTypeClass;
-  private final Class<? extends InputStorageType> inputStorageTypeClass;
+  private final Class<ArgType> argTypeClass;
+  private final Class<? extends ArgStorageType> argStorageTypeClass;
 
-  protected abstract Builder createOutputBuilder(int size);
+  protected Builder createOutputBuilder(int size) {
+    return new ObjectBuilder(size);
+  }
```

Comment:
I don't think we should have this default implementation, it can be error prone.

Note that `ObjectBuilder` always creates columns of type `Mixed`. That's rather rarely what we want here.

Better default could be `InferredBuilder` that will infer the type. But that is still wrong, because most of the time we know the output type and so we'd rather use a specialized builder for better efficiency.

IMO this should be implemented by each specialization.

Reply:
I must admit I didn't dig into what this actually did. Prior to this PR there were only 2 spots that this was overridden and both used the above code. Thanks for the additional information: I'll take a proper look.

Reply:
The previous 2 usages used `ObjectBuilder`, because they were returning the `Duration` type which does not have a corresponding `Value_Type`.

It could get a more precise type if we implement #7264 or maybe #6163

Reply:
I reverted all my changes to this file and went back to my std-bits/table/src/main/java/org/enso/table/data/column/operation/map/text/StringStringLongOp.java solution. I am seeing now why the specialised string versions existed.